feature_engine/encoding/base_encoder.py

```diff
@@ -183,6 +195,13 @@ def transform(self, X: pd.DataFrame) -> pd.DataFrame:
                 else:
                     X[feature] = X[feature].astype("float")
 
+        self._check_nas_in_result(X)
+        return X
+
+    def _check_nas_in_result(self, X):
```

Comment:
this method is only used in the new class, correct? can we not place it there instead?

Reply:
this method needs to be moved to MatchVariables